### PR TITLE
Handles

### PR DESCRIPTION
**Pull request by toledoal**
Hey! 😄 

This PR adds a selection event for the Keyframe. It also displays the handles. 
The handles are now generated from the inTangent and outTangent values from the animation. 
To use the handles the user needs to click and select the handle and move it up and down to give the correct inTangent, outTangent values. 

selectHandle
 

I still need to add the "broken mode" to the handles. Currently the opposite handle is attached to the selected handle.

### CODE REVIEW

File: inspector/src/components/actionTabs/tabs/propertyGrids/animations/animationCurveEditorComponent.tsx [hunk 1]
```diff
@@ -504,32 +576,90 @@ export class AnimationCurveEditorComponent extends React.Component<IAnimationCur
...
+                let prev = this.state.svgKeyframes?.find(kf => kf.id === i.toString());
+                if (prev){
+                svgKeyframe.isLeftActive = prev?.isLeftActive;
+                svgKeyframe.isRightActive = prev?.isRightActive;
+                svgKeyframe.selected = prev?.selected
+                }
+
+
+            }
+
```
Comment (sebavan) [inline inspector/src/components/actionTabs/tabs/propertyGrids/animations/animationCurveEditorComponent.tsx:654]: A bit too many spaces here :-)

File: inspector/src/components/actionTabs/tabs/propertyGrids/animations/animationCurveEditorComponent.tsx [hunk 2]
```diff
@@ -504,32 +576,90 @@ export class AnimationCurveEditorComponent extends React.Component<IAnimationCur
...
+                } else if (inTangent !== null) {
+                    data += ` ${inTangent.x} ${inTangent.y} ${svgKeyframe.keyframePoint.x} ${svgKeyframe.keyframePoint.y} `
                 }
 
             }
 
+            if (this.state){
+                let prev = this.state.svgKeyframes?.find(kf => kf.id === i.toString());
+                if (prev){
+                svgKeyframe.isLeftActive = prev?.isLeftActive;
```
Comment (sebavan) [inline inspector/src/components/actionTabs/tabs/propertyGrids/animations/animationCurveEditorComponent.tsx:647]: indentation

File: inspector/src/components/actionTabs/tabs/propertyGrids/animations/animationCurveEditorComponent.tsx [hunk 3]
```diff
@@ -621,12 +751,60 @@ export class AnimationCurveEditorComponent extends React.Component<IAnimationCur
...
+
+
         animation.setKeys(keys);
 
         this.selectAnimation(animation);
 
     }
 
+    rotateTangent(direction: number, updatedKeyframe: IKeyframeSvgPoint){
+
```
Comment (sebavan) [inline inspector/src/components/actionTabs/tabs/propertyGrids/animations/animationCurveEditorComponent.tsx:805]: It does not look like it is used ?

Comment (toledoal) [reply inspector/src/components/actionTabs/tabs/propertyGrids/animations/animationCurveEditorComponent.tsx:805]: good catch, thanks!

**Comment by sebavan** [comment]
Overall looks ok but in progress ? I let you ping me once done and I ll go thoroughly on it ?

**Comment by toledoal** [comment]
Sure, thank you. I still want to refactor this code 😄

**Comment by toledoal** [comment]
Hello, @sebavan  I fixed the pending items I had here and also checked the code format. Thank you for your review 😄

**Comment by sebavan** [comment]
This looks all good to me, will just request an extra review from @deltakosh which will be back tomorrow :-)

**Comment by toledoal** [comment]
That sounds good! Thanks!

**Comment by deltakosh** [comment]
LGTM! thanks!

**Pull request merged**